### PR TITLE
Feature/pim 1776 sync product data uw api

### PR DESCRIPTION
**Pull request by dpitk2222**

### CODE REVIEW

File: cirras-underwriting-api/cirras-underwriting-api-rest-endpoints/src/test/java/ca/bc/gov/mal/cirras/underwriting/api/rest/v1/endpoints/ProductEndpointTest.java [hunk 1]
```diff
@@ -0,0 +1,371 @@
...
+		service.synchronizeProduct(resource2);
+		
+		ProductRsrc fetchedResource2 = service.getProduct(topLevelEndpoints, productId.toString()); 
+
+		Assert.assertEquals(resource2.getProductId(), fetchedResource2.getProductId());
+
+		//DELETE PRODUCT using the other delete method.
+		service.deleteProduct(topLevelEndpoints, productId.toString());
+
+		ProductRsrc deletedResource2 = service.getProduct(topLevelEndpoints, productId.toString());
```
Comment (mhostett) [inline cirras-underwriting-api/cirras-underwriting-api-rest-endpoints/src/test/java/ca/bc/gov/mal/cirras/underwriting/api/rest/v1/endpoints/ProductEndpointTest.java:253]: Shouldn't this return a 404 not found error because the resource doesn't exist anymore?

File: cirras-underwriting-api/cirras-underwriting-service-api/src/main/java/ca/bc/gov/mal/cirras/underwriting/service/api/v1/impl/CirrasDataSyncServiceImpl.java [hunk 2]
```diff
@@ -630,6 +639,117 @@ public void deletePolicy(Integer policyId,
...
+		Product result = null;
+
+		try {
+			ProductDto dto = productDao.fetch(productId);
+
+			if (dto != null) {
+				result = cirrasDataSyncFactory.getProduct(dto);
+			}
+			else {
+				throw new NotFoundException("Did not find product record: productId: " + productId);
```
Comment (mhostett) [inline cirras-underwriting-api/cirras-underwriting-service-api/src/main/java/ca/bc/gov/mal/cirras/underwriting/service/api/v1/impl/CirrasDataSyncServiceImpl.java:658]: Other sync get services don't return a not found exception because it's only used for testing.

**Review by mhostett — CHANGES_REQUESTED** [review]

File: cirras-underwriting-api/cirras-underwriting-persistence/src/test/java/ca/bc/gov/mal/cirras/underwriting/persistence/v1/dao/ProductDaoTest.java [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+		createPolicy(policyId1, growerId1, contractId1, cropYear1, policyNumber1, contractNumber1, insurancePlanId, policyStatusCode, officeId);
+		
+
+		ProductDao dao = persistenceSpringConfig.productDao();
+		
+		ProductDto newDto = new ProductDto();
+
+		newDto.setCommodityCoverageCode("CQG");
+		newDto.setCropCommodityId(16);
+		newDto.setDeductibleLevel(20);
```
[the remark this answers is not in the document]
Comment (dpitk2222) [reply cirras-underwriting-api/cirras-underwriting-persistence/src/test/java/ca/bc/gov/mal/cirras/underwriting/persistence/v1/dao/ProductDaoTest.java:178]: Good point, we haven't really been doing that consistently, but I'll start that for future implementations. Like maybe we need two update tests, one with all non-null values, and one with null values for nullable fields.

**Review by mhostett — APPROVED** [review]
